manager/allocator/cnmallocator/drivers.go

```diff
@@ -0,0 +1,25 @@
+package cnmallocator
+
+import (
+	"fmt"
+
+	"github.com/docker/docker/libnetwork/driverapi"
+	"github.com/docker/docker/libnetwork/drivers/remote"
+)
+
+type driverRegisterFn func(r driverapi.Registerer, config map[string]interface{}) error
+
+func registerRemote(r driverapi.Registerer, _ map[string]interface{}) error {
+	dc, ok := r.(driverapi.DriverCallback)
+	if !ok {
+		return fmt.Errorf(`failed to register "remote" driver: driver does not implement driverapi.DriverCallback`)
+	}
+	return remote.Register(dc, dc.GetPluginGetter())
+}
```

Comment:
> so that Swarmkit can stop depending on the deprecated drvregistry.DrvRegistry. 

I need to check that last one; while working on https://github.com/moby/swarmkit/pull/3139 (reviews welcome on that one as well 😅), I found that the `plugingetter` field is not exported, but looks to be used, so the only way to set it is through the (deprecated) `drvregistry.New`

I may need to have a deeper look "how" it's used, and if there's other ways to pass it around though.

Reply:
plugingetter field? `cnmallocator.New` takes the PluginGetter as an argument, constructs a `DrvRegistry` using it, and passes it to each the `initializers` in the slice/map. Simply cut out the middleman by passing the PluginGetter directly into a call to `remote.Register` from `cnmallocator.New`. That's what I meant by special-casing the registration.

Reply:
Ah! Gotcha (I think) I'll have another look tomorrow

Reply:
You're right that the `DrvRegistry.pluginGetter` field is used. It serves one purpose: implementing the `GetPluginGetter()` method, which is only used to indirectly pass the PluginGetter into `remote.Init()`. As I explained in the commit message of https://github.com/moby/moby/commit/5595311209cc915e8b0ace0a1bbd8b52a7baecb0, I very deliberately omitted any PluginGetter stuff from the replacement types.